### PR TITLE
OZW emulator

### PR DESCRIPTION
**Pull request by marcelveldt**
not sure this needs review as it's just a script helpful for debugging.

### CODE REVIEW

File: script/ozw_emulator.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    "timeout-disconnect-delay": 2,
+    "auth": {"allow-anonymous": "true", "plugins": ["auth.anonymous"]},
+    "topic-check": {"enabled": True, "plugins": ["topic_taboo"]},
+}
+
+
+def get_args():
+    """Get arguments."""
+    parser = argparse.ArgumentParser(description="OZW Emulator")
+    parser.add_argument("filename", type=str, help="File with dump from mqtt.")
```
Comment (MartinHjelmare) [inline script/ozw_emulator.py:35]: Can we add a broker host option?

Comment (MartinHjelmare) [reply script/ozw_emulator.py:35]: Wait, we're running the broker too. So then it's not needed. Missed that.

Comment (marcelveldt) [reply script/ozw_emulator.py:35]: Yeah, it's all-in-one, It runs a local broker and publishes all topics from the dump there..

Comment (marcelveldt) [reply script/ozw_emulator.py:35]: But adding an option to use another broker is possible too offcourse

File: script/ozw_emulator.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+async def publish_from_file():
+    """Publish all topics from the provided dump file."""
+    with open(dump_file, "rt") as fp:
+        for line in fp:
+            topic, payload = line.strip().split(",", 1)
+            await mqtt_client.publish(topic, payload.encode(), retain=True)
+
+
+async def process_messages():
```
Comment (MartinHjelmare) [inline script/ozw_emulator.py:47]: Maybe accept the client here to not depend on the global scope.

File: script/ozw_emulator.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+
+def get_args():
+    """Get arguments."""
+    parser = argparse.ArgumentParser(description="OZW Emulator")
+    parser.add_argument("filename", type=str, help="File with dump from mqtt.")
+    return parser.parse_args()
+
+
+async def publish_from_file():
```
Comment (MartinHjelmare) [inline script/ozw_emulator.py:39]: Accept the dump file and client here.

File: script/ozw_emulator.py (hunk 2 above)
Comment (marcelveldt) [reply script/ozw_emulator.py]: done

File: script/ozw_emulator.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            for value_topic in mqtt_data:
+                if value_topic.endswith(f'/value/{data["ValueIDKey"]}/'):
+                    payload = mqtt_data[value_topic]
+                    payload["Value"] = value
+                    payload = json.dumps(payload).encode()
+                    await mqtt_client.publish(value_topic, payload, retain=True)
+                    break
+
+
+async def run(event_loop):
```
Comment (MartinHjelmare) [inline script/ozw_emulator.py:65]:
```suggestion
async def run(event_loop, args):
```

File: script/ozw_emulator.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+if __name__ == "__main__":
+
+    args = get_args()
+    formatter = "[%(asctime)s] :: %(levelname)s :: %(name)s :: %(message)s"
+    logging.basicConfig(level=logging.INFO, format=formatter)
+    loop = asyncio.get_event_loop()
+
+    # Run
+    loop.create_task(run(loop))
```
Comment (MartinHjelmare) [inline script/ozw_emulator.py:101]:
```suggestion
    loop.create_task(run(loop, args))
```

**Comment by marcelveldt** [comment]
I hope the script can be helpful, I've used it today myself to debug a users having problems.

**Pull request merged**